packages/map-template/src/components/Search/Categories/Categories.jsx

```diff
@@ -151,19 +159,60 @@ function Categories({ onSetSize, getFilteredLocations, searchFieldRef, isOpen })
         setCategoriesWithoutChildKeys(topLevelCategories);
     }, [categories]);
 
+    /**
+     * Collects all child keys from all categories.
+     * Filters out categories from the initial view that are not listed as a child to another category.
+     */
+    useEffect(() => {
+        const childKeys = categories.flatMap(([, category]) => category.childKeys || []);
+        const childLevelCategories = categories.filter(([key]) => childKeys.includes(key));
+        setCategoriesWithChildKeys(childLevelCategories);
+    }, [categories]);
+
+    /**
+     * Gets display name of selected category.
+     */
+    useEffect(() => {
+        const categoryDisplayName = categories.find(([key]) => key === selectedCategory)?.[1]?.displayName;
+        setSelectedCategoryDisplayName(categoryDisplayName)
+    }, [categories])
+
     return (
         <div className="categories prevent-scroll" {...scrollableContentSwipePrevent}>
             {categories.length > 0 && (
                 <div className="categories__list">
-                    {categoriesWithoutChildKeys
-                        .map(([category, categoryInfo]) => (
+                    {topLevelCategory ? (
+                        categoriesWithoutChildKeys.map(([category, categoryInfo]) => (
                             <div key={category} className="categories__category">
                                 <button onClick={() => categoryClicked(category)}>
                                     <img src={categoryInfo.iconUrl} alt="" />
                                     {categoryInfo.displayName}
                                 </button>
                             </div>
-                        ))}
+                        ))
+                    ) : (
+                        <>
+                            <div className="categories__nav">
+                                <button
+                                    aria-label="Back"
```

Comment:
We must use the translation function `t()` for textual content that is not coming from MapsIndoors data.